scripts/certbot/godaddy_hook.py

```
@@ -0,0 +1,134 @@
+    """Return optional environment variable *name* with *default*."""
+
+    return (os.environ.get(name) or default).strip()
+
+
+def _zone_and_name(fqdn: str) -> tuple[str, str]:
+    """Split ACME record fqdn into GoDaddy zone domain and relative host name."""
+
+    value = fqdn.rstrip(".")
+    labels = [part for part in value.split(".") if part]
+    if len(labels) < 2:
+        raise RuntimeError(f"Invalid DNS name for ACME challenge: {fqdn}")
```

Ruff (0.14.14)

[warning] Avoid specifying long messages outside the exception class

(TRY003)